### PR TITLE
Assimp shift

### LINKED ISSUE #76: OBJLoader is not handling certain objects well
Try loading this: https://free3d.com/3d-model/alien-animal-update-in-blender-28x-953251.html
This is how it looks in Rootex
image
This is how it should look like: 
image

### PR DESCRIPTION
**Pull request by allrounder27**
Fixes #76
Add Assimp as the new object loading library replacing OBJ-Loader

### CODE REVIEW

File: rootex/core/resource_loader.cpp [hunk 1]
```diff
@@ -169,30 +172,48 @@ VisualModelResourceFile* ResourceLoader::CreateVisualModelResourceFile(const Str
...
+	PANIC(scene == false, "Model could not be loaded: " + OS::GetAbsolutePath(path).generic_string());
 
 	VertexData vertex;
 
 	Vector<VertexData> vertices;
-	vertices.reserve(s_ModelLoader.LoadedVertices.size());
+	vertices.reserve(mesh->mNumVertices);
 
-	for (auto& v : s_ModelLoader.LoadedVertices)
+	for (unsigned int v = 0; v < mesh->mNumVertices; v++)
```
Comment (sin3point14) [inline rootex/core/resource_loader.cpp:191]: use range based for loop

File: rootex/core/resource_loader.cpp [hunk 2]
```diff
@@ -169,30 +172,48 @@ VisualModelResourceFile* ResourceLoader::CreateVisualModelResourceFile(const Str
...
+		vertex.m_Normal.y = mesh->mNormals[v].y;
+		vertex.m_Normal.z = mesh->mNormals[v].z;
+		vertex.m_TextureCoord.x = mesh->mTextureCoords[0][v].x;
+		vertex.m_TextureCoord.y = mesh->mTextureCoords[0][v].y;
 		vertices.push_back(vertex);
 	}
 
+	std::vector<unsigned short> Indices;
+
+	for (unsigned int f = 0; f < mesh->mNumFaces; f++)
```
Comment (sin3point14) [inline rootex/core/resource_loader.cpp:206]: range based for loop here as well

File: rootex/core/resource_loader.cpp [hunk 3]
```diff
@@ -169,30 +172,48 @@ VisualModelResourceFile* ResourceLoader::CreateVisualModelResourceFile(const Str
...
+		vertex.m_Position.z = mesh->mVertices[v].z;
+		vertex.m_Normal.x = mesh->mNormals[v].x;
+		vertex.m_Normal.y = mesh->mNormals[v].y;
+		vertex.m_Normal.z = mesh->mNormals[v].z;
+		vertex.m_TextureCoord.x = mesh->mTextureCoords[0][v].x;
+		vertex.m_TextureCoord.y = mesh->mTextureCoords[0][v].y;
 		vertices.push_back(vertex);
 	}
 
+	std::vector<unsigned short> Indices;
```
Comment (sin3point14) [inline rootex/core/resource_loader.cpp:204]: Indices -> indices

File: rootex/core/resource_loader.cpp (hunk 1 above)
Comment (allrounder27) [reply rootex/core/resource_loader.cpp:191]: but here the loop is iterating over three different arrays

File: rootex/core/resource_file.cpp [hunk 4]
```diff
@@ -193,29 +193,47 @@ VisualModelResourceFile::~VisualModelResourceFile()
 void VisualModelResourceFile::reload()
```
[the remark this answers is not in the document]
Comment (allrounder27) [reply rootex/core/resource_file.cpp:193]: done

File: rootex/core/resource_loader.cpp [hunk 5]
```diff
@@ -169,30 +172,50 @@ VisualModelResourceFile* ResourceLoader::CreateVisualModelResourceFile(const Str
...
+	      aiProcess_CalcTangentSpace      |
+		  aiProcess_Triangulate           | 
+		  aiProcess_JoinIdenticalVertices | 
+		  aiProcess_SortByPType);
+    const aiMesh* mesh = scene->mMeshes[0]; //Taking single mesh
+	aiFace* face;
+
 	// File not found in cache, load it only once
-	PANIC(s_ModelLoader.LoadFile(OS::GetAbsolutePath(path).generic_string()) == false, "Model could not be loaded: " + OS::GetAbsolutePath(path).generic_string());
+	PANIC(scene == false, "Model could not be loaded: " + OS::GetAbsolutePath(path).generic_string());
```
Comment (IronicallySerious) [inline rootex/core/resource_loader.cpp:184]: Compare with nullptr instead of false. It is unidiomatic to compare pointers with bool directly

**Comment by IronicallySerious** [comment]
LGTM

**Pull request merged**